samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -881,4 +919,23 @@ void BLEConnectivityManager::Recovery::StartTimer()
 	k_timer_start(&mRecoveryTimer, K_SECONDS(time), K_NO_WAIT);
 }
 
+BLEConnectivityManager::State BLEConnectivityManager::GetCurrentState()
+{
+	for (uint8_t s = State::Scanning; s <= State::Idle; s++) {
+		if (mStateBitmask & (1 << s)) {
+			return static_cast<State>(s);
+		}
+	}
+	return State::Unknown;
+}
+
+void BLEConnectivityManager::UpdateStateFlag(State newState, bool enabled)
```

Comment:
Should we deliberately treat some states as mutually exclusive?

Reply:
I would like to do it as easily as possible, so I decided to support only the current state. Moreover, we can manage only the current action in the BLE Connectivity Manager, so we don't need to signalize more states simultaneously. We can discuss it f2f.